google/cloud/storage/options.h

```diff
@@ -64,6 +65,11 @@ struct CAPathOption {
   using Type = std::string;
 };
 
+/// This is only intended for testing of the library. Not for public use.
+struct RestConfiguration {
+  using Type = std::string;
```

Comment:
Q.  I see this is set-to/checked-as "rest", which seems redundant ... RestConfiguration == "rest".  Perhaps it should be "TransportOption", or something like that?

Reply:
Turned this into a boolean.